src/trust_is_risk.js

```diff
@@ -0,0 +1,282 @@
+// @flow
+import type {Entity, TXHash, Key} from "./types";
+var bcoin = require("bcoin");
+var Address = bcoin.primitives.Address;
+var KeyRing = bcoin.primitives.KeyRing;
+var MTX = bcoin.primitives.MTX;
+var Input = bcoin.primitives.Input;
+var Output = bcoin.primitives.Output;
+var Outpoint = bcoin.primitives.Outpoint;
+var Coin = bcoin.primitives.Coin;
+var assert = require("assert");
+var helpers = require("./helpers");
+var TrustDB = require("./trust_db");
+var DirectTrust = require("./direct_trust");
+
+class TrustIsRisk {
+  node : bcoin$FullNode
+  db : TrustDB
+
+  constructor(node : bcoin$FullNode) {
+    this.node = node;
+    this.db = new TrustDB();
+
+    this.node.on("tx", this.addTX.bind(this));
+  }
+
+  getTrust(origin : Entity, dest : Entity) {
+    return this.db.getTrustAmount(origin, dest);
+  }
+
+  getDirectTrust(origin : Entity, dest : Entity) {
+    return this.db.getDirectTrustAmount(origin, dest);
+  }
+
+  // Attempts to parse a bitcoin transaction as a trust change and adds it to the trust network
+  // if successful.
+  // Returns true if the transaction is a TIR transaction and was successfully added to the
+  // network, false otherwise.
+  // Throws an error if the transaction was processed earlier.
+  addTX(tx : bcoin$TX) : boolean {
+    var txHash = tx.hash().toString("hex");
+    if (this.db.isTrustTX(txHash)) {
+      throw new Error(`Transaction already processed: Transaction ${txHash} already carries trust`);
+    }
+
+    var directTrusts = this.getDirectTrusts(tx);
+    if (directTrusts.length === 0) return false;
+    else {
```

Comment:
I disagree, I think else after return is fine here. The if is not guarding against something. Here I want to distinguish between two cases and the else makes the code more readable.